### DISCUSSION TITLE
Adaptive CRT shaders

### DISCUSSION THREAD

**Discussion by roberttryton** [dosbox-staging/dosbox-staging#3577] [Ideas]
Whether the black lines appear or not depends on size of monitor, resolution and brightness/contrast settings.
if i remember correctly 14" screens doesn't even have them in 320x200.
larger monitors use quad scanning for lower resolutions and double scanning for middle ones to get rid those black lines.
higher brightness/contrast settings make black lines thinner and finally make them disappear.
lower brightness/contrast settings may make black lines appear and make them thicker.
if you have an old burned out screen that is not able to reach its original brightness/contrast levels that obviously also affects them too 

CRTs kind of had their optimal resolution ranges (two or three) just like LCDs have their one native resolution.
I wonder if someone makes LCD shaders for non-native LCDs resolutions for high resolution screens ;)

**Comment by johnnovak** [comment]
Yes, and we do proper VGA double scanning like you described and the results mimic CRTs very closely. I tweaked the shaders using my real CRTs.

"Quad scanning" does not exist...

The brightness/contrast variable scanline thing might be implemented in the future when we introduce monitor controls.

Not too sure—is there a question here?

**Comment by roberttryton** [comment]
(no question in the previous post)

what CRT sizes have you tested?

**Comment by roberttryton** [comment]
ok, so i tested a 19" and I think recalled 15" and 14" tests

so yes quad not exist,
brightness/contrast doesn't affect black lines in a noticeable way on a 19"
(it was an issue on an amber 9-pin screen)

19" doesn't have black lines in 512x384 and 1024x768 and higher
15" doesn't have black lines in 320x240 and 640x480 and higher
14" doesn't have black lines in 320x200 and 640x400 and higher

**Comment by johnnovak** [comment]
I have 15" and 17" CRTs here.
